source/running-simulation.md

```diff
@@ -0,0 +1,187 @@
+4. Calculate the [housing_allowance](https://demo.openfisca.org/legislation/housing_allowance) and print its value for every test case household.
+
+Which gives:
+
+```python
+# -*- coding: utf-8 -*-
+
+from openfisca_core.simulation_builder import SimulationBuilder
+from openfisca_country_template import CountryTaxBenefitSystem
+
+
+TEST_CASE = {
+    # ... whole test case ...
+}
+
+tax_benefit_system = CountryTaxBenefitSystem()
+
+simulation_builder = SimulationBuilder()
+simulation = simulation_builder.build_from_entities(tax_benefit_system, TEST_CASE)
+
+housing_allowance = simulation.calculate('housing_allowance', '2011-01')
+
+print("households", simulation.household.ids)
+print("housing_allowance", housing_allowance)
+```
+
+## How to run a simulation on data
+
+You can build a *Simulation* on multiple data formats.
+Any well structured tabular input shoud be fine as long as you are able to iterate over its items in Python.
+
+In the following example, will use the [pandas](https://pandas.pydata.org) library to do so.
+
+### CSV data
+
+To apply the legislation on data described in one or more CSV files, you can use the OpenFisca Python API.
+
+#### Application: calculate a population's income tax from a CSV file
+
+Let's say you are using the [country-template](https://github.com/openfisca/country-template), which describes the legislation of a yet to be country.
+
+Let's also say you have the following `data.csv` and that you want to calculate [income_tax](https://demo.openfisca.org/legislation/income_tax) for all persons:
+
+```csv
+person_id,person_salary,person_age
+1,2694,40
+2,2720,43
+3,1865,45
+4,1941,23
+5,2393,31
+6,3008,47
+7,2286,23
+8,3386,28
+9,2929,38
+10,3981,37
+11,3643,38
+12,2078,23
+```
+
+1. Install the required libraries, by running in your console:
+
+```sh
+$ python --version # Python 3.7.0 or greater should be installed on your computer
+$ pip install --upgrade pip openfisca_country_template pandas ipython
+$ ipython
+```
+
+2. Load the `country-template` legislation and, then, the content of the `data.csv` file with the [pandas](https://pandas.pydata.org) library:
+
+```python
+In [1]: from openfisca_country_template import CountryTaxBenefitSystem
+
+In [2]: import pandas as pandas
+
+In [3]: tax_benefit_system = CountryTaxBenefitSystem()
+
+In [4]: data = pandas.read_csv('./data.csv')  # pandas.DataFrame object
+
+In [5]: length = len(data)  # ignores CSV header
+```
+
+You can now access the `person_salary` column values:
+
+```python
+In [6]: data.person_salary
+Out[6]:
+0     2694
+1     2720
+2     1865
+3     1941
+4     2393
+5     3008
+6     2286
+7     3386
+8     2929
+9     3981
+10    3643
+11    2078
+Name: person_salary, dtype: int64
+```
+
+
+3. Build a simulation according to your data's length:
+
+```python
+In [7]: from openfisca_core.simulation_builder import SimulationBuilder
+
+In [8]: simulation = SimulationBuilder().build_default_simulation(tax_benefit_system, length)
```

Comment:
This is not consistent with what is documented in [this PR](https://github.com/openfisca/openfisca-core/pull/833) 🤔. I think I'd feel more confortable with merging the latter first, and have a doc that really reflects how we recommend to initialize CSV simulations.

Reply:
Yes (and PR on CSV merged just now :tada:). Nevertheless, as this PR only moves this section on CSV data, I would be in favour of refactoring the example in an other PR.